### PR TITLE
Replace --test_component_name argument with internlal __lb_name

### PR DESCRIPTION
**Pull request by ewiseblatt**
@lwander

### CODE REVIEW

File: spinnaker/spinnaker_system/aws_smoke_test.py [hunk 1]
```diff
@@ -102,13 +96,17 @@ def __init__(self, bindings, agent=None):
...
-            app=bindings['TEST_APP'],
-            stack=bindings['TEST_STACK'],
-            detail=bindings['TEST_COMPONENT_DETAIL']))
+
+    optional_stack = ('-{stack}'.format(stack=bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+
+    self.__lb_detail = 'lb'
+    self.__lb_name = '{app}{optional_stack}-{detail}'.format(
```
Comment (lwander) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: Pretty sure if you omit stack, you're left with two `-` characters. This should reflect that

File: spinnaker/spinnaker_system/aws_smoke_test.py [hunk 2]
```diff
@@ -326,12 +324,15 @@ def create_server_group(self):
...
 
     # Spinnaker determines the group name created,
     # which will be the following:
-    group_name = '{app}-{stack}-v000'.format(
-        app=self.TEST_APP, stack=bindings['TEST_STACK'])
+    optional_stack = ('-{stack}'.format(stack=bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+
+    group_name = '{app}{optional_stack}-v000'.format(
```
Comment (lwander) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: Why doesn't this have detail?

File: spinnaker/spinnaker_system/aws_smoke_test.py (hunk 1 above)
Comment (lwander) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: The exception is when you have no stack or detail filled, then it's just `{app}`. See the formatting logic here: https://github.com/Netflix/frigga/blob/master/src/main/java/com/netflix/frigga/NameBuilder.java

File: spinnaker/spinnaker_system/aws_smoke_test.py [hunk 3]
```diff
@@ -396,8 +397,12 @@ def delete_server_group(self):
     is no longer visible on AWS (or is in the process of terminating).
     """
     bindings = self.bindings
-    group_name = '{app}-{stack}-v000'.format(
-        app=self.TEST_APP, stack=bindings['TEST_STACK'])
+
+    optional_stack = ('-{stack}'.format(stack=bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+    group_name = '{app}{optional_stack}-v000'.format(
```
Comment (lwander) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: See above

File: spinnaker/spinnaker_system/google_smoke_test.py [hunk 4]
```diff
@@ -94,13 +89,17 @@ def __init__(self, bindings, agent=None):
...
-            app=bindings['TEST_APP'],
-            stack=bindings['TEST_STACK'],
-            detail=bindings['TEST_COMPONENT_DETAIL']))
+
+    optional_stack = ('-{stack}'.format(stack=bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+
+    self.__lb_detail = 'lb'
+    self.__lb_name = '{app}{optional_stack}-{detail}'.format(
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_smoke_test.py]: See above

File: spinnaker/spinnaker_system/aws_smoke_test.py (hunk 1 above)
Comment (ewiseblatt) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: good catch. you need the empty stack or detial would be ambiguous.

Comment (ewiseblatt) [inline spinnaker/spinnaker_system/aws_smoke_test.py]: fixed

**Comment by lwander** [comment]
LGTM

**Pull request merged**